Model/Serverside/Request/Frontend.php

```diff
@@ -0,0 +1,274 @@
+    {
+        $this->request->addEvent(
+            'page_view',
+            isset($params['page']) ? $params['page'] : []
+        );
+    }
+
+    /**
+     * Adds Product View event to the request
+     *
+     * @param mixed[] $params
+     */
+    public function addProductView($params)
+    {
+        if (isset($params['product']) && !empty($params['product'])) {
+            $this->request->addEvent(
+                'product_view',
+                ['id' => $params['product']['Id']]
+            );
+        }
+    }
+
+    /**
+     * Adds Customer details event to the request (if needed)
+     * @param integer $storeId
+     */
+    public function addCustomerDetails($storeId)
+    {
+        $logout = false;
+        $customerDetailsCheck = $this->customer->checkCustomer();
+        if ($customerDetailsCheck['send']) {
+            if (!empty($customerDetailsCheck['details'])) {
+                $this->request->addEvent(
+                    'customer_details',
+                    $customerDetailsCheck['details']
+                );
+            } else {
+                $this->cookie->customerLogout($storeId);
+                $logout = true;
+            }
+        }
+
+        if ($logout === false) {
+            $pcv = $this->cookie->getCookie('pc_v_', $storeId);
+            if ($pcv) {
+                $this->request->setVisitorId($pcv);
+            }
+
+            $pcSessid = $this->cookie->getCookie('pc_sessid_', $storeId);
+            if ($pcSessid) {
+                $this->request->setSessionId($pcSessid);
+            }
+        }
+    }
+
+    /**
+     * Adds the set basket event to the request (if needed)
+     */
+    public function addSetBasket()
+    {
+        $cartCheck = $this->cart->checkCart();
+        if ($cartCheck['send']) {
+            $this->request->addEvent(
+                'set_basket',
+                $cartCheck['items']
+            );
+        }
+    }
+
+    /**
+     * Adds the order event if needed
+     * @param mixed[] $params
+     */
+    public function addOrder($params)
+    {
+        if (isset($params['order']) && !empty($params['order'])) {
+            $this->request->addEvent('order_track', $params['order']);
+        }
+    }
+
+    /**
+     * Processes the result from PureClarity
+     *
+     * Renders Zones and updates cookies
+     *
+     * @param Data $request
+     * @param string $storeId
+     * @return mixed[]
+     */
+    public function processResult($request, $storeId)
+    {
+        $result = $this->serverSide->getResult();
+
+        if (isset($result['zones'])) {
+            $this->productData->setCurrentUrl($request->getCurrentUrl());
+            foreach ($result['zones'] as $zoneId => $zone) {
+                $result['zones'][$zoneId]['items'] = $this->productData->getProductData($zone);
+                unset($result['zones'][$zoneId]['data']['items']);
+            }
+        }
+
+        if (isset($result['visitorId'])) {
```

Comment:
second place I've seen the expiry for these cookies. Should both be constants somewhere